### ISSUE TITLE
Index can remain locked on filesystem, impossible to unlock it

### ISSUE THREAD

**Issue by pavelbazika** [blevesearch/bleve#1473]
In function `bleve.openIndexUsing`, when an error occurs after successful call of `rv.i.Open()`, there is no possibility to close already opened index, because `bleve.openIndexUsing` returns nil index reference in such case and also does not close the index by itself. When the index is not opened in readonly mode, flock remains locked on index file, preventing future open attempts.

Such situation can happen when one of these calls in `bleve.openIndexUsing` fail:
* `indexReader, err := rv.i.Reader()`
* `mappingBytes, err := indexReader.GetInternal(mappingInternalKey)`
* `err = json.Unmarshal(mappingBytes, &im)`

Expected behavior: `bleve.openIndexUsing` should either return opened index or close it and return nil.

The issue happened in debug environment, when one of the scorch zap files was created by user root, but root.bolt was created by plain user. Then root.bolt is loaded successfuly, but loading one of the zap files fails with message:
```
unable to load snapshot, failed to load segment: error opening bolt segment: open /path/to/index/store/0000000000f9.zap: permission denied, continuing
```
and no error is returned to `bleve.openIndexUsing` so it fails later. I admit this is unusual situation caused by debugging, when part of the index was owned by another user, but please review if not returning an error from zap to bleve is intended behavior in such case.

Thanks

**Comment by pavelbazika** [comment]
Proposed fix: https://github.com/blevesearch/bleve/pull/1478

**Comment by eslindsey** [comment]
I believe this should be closed now? Looks like the fix in #1479 is merged.
